# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1237938116 14400
#      Tue Mar 24 19:41:56 2009 -0400
# Node ID dd5725531732d033fa4d39e772eaa2f67744097e
# Parent  f914834836e75c63bcf99af360eb1e9459e1b48d
gnuplot_drawnow.m: When printing, pass scalar plot_stream to __gnuplot_draw_figure__, and close all plot streams when done.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-03-24  Ben Abbott <bpabbott@mac.com>
+
+	* plot/gnuplot_drawnow.m: When printing, pass scalar plot_stream
+	to __gnuplot_draw_figure__, and close all plot streams when done.
+
 2009-03-24  John W. Eaton  <jwe@octave.org>
 
 	* general/isa.m: Handle parent classes.
 
 2009-03-23  Ben Abbott <bpabbott@mac.com>
 
 	* plot/gnuplot_drawnow.m: Check that gnuplot has internal variable
 	"GPVAL_TERMINALS".
diff --git a/scripts/plot/gnuplot_drawnow.m b/scripts/plot/gnuplot_drawnow.m
--- a/scripts/plot/gnuplot_drawnow.m
+++ b/scripts/plot/gnuplot_drawnow.m
@@ -47,29 +47,32 @@ function gnuplot_drawnow (h, term, file,
         gnuplot_supports_term = any (strcmpi (available_terminals,
                                               gnuplot_trim_term (term)));
       else
         gnuplot_supports_term = true;
       endif
       if (gnuplot_supports_term)
         [enhanced, implicit_margin] = gnuplot_set_term (plot_stream (1), true,
                                                         h, term, file);
-        __go_draw_figure__ (h, plot_stream, enhanced, mono, printing, implicit_margin);
+        __go_draw_figure__ (h, plot_stream(1), enhanced, mono, printing, implicit_margin);
         if (nargin == 5)
           fid = fopen (debug_file, "wb");
           [enhanced, implicit_margin] = gnuplot_set_term (fid, true, h, term, file);
           __go_draw_figure__ (h, fid, enhanced, mono, printing, implicit_margin);
         endif
       else
         error ("gnuplot_drawnow: the gnuplot terminal, \"%s\", is not available.",
                gnuplot_trim_term (term))
       endif
     unwind_protect_cleanup
       if (! isempty (plot_stream))
-        pclose (plot_stream);
+        pclose (plot_stream(1));
+        if (numel (plot_stream) == 2)
+          pclose (plot_stream(2));
+        endif
       endif
       if (! isempty (fid))
         fclose (fid);
       endif
     end_unwind_protect
   elseif (nargin == 1)
     ##  Graphics terminal for display.
     plot_stream = get (h, "__plot_stream__");
